synapse/rest/media/v1/thumbnail_resource.py

```diff
@@ -43,11 +43,11 @@ def _async_render_GET(self, request):
         m_type = parse_string(request, "type", "image/png")
 
         if server_name == self.server_name:
-            yield self._respond_local_thumbnail(
+            yield self._select_or_generate_local_thumbnail(
                 request, media_id, width, height, method, m_type
             )
         else:
-            yield self._respond_remote_thumbnail(
+            yield self._select_or_generate_remote_thumbnail(
                 request, server_name, media_id,
```

Comment:
Presumably this will switch between _respond_remote_thumbnail and _select_or_generate_remote_thumbnail when the config option is added.